src/utils.ts

```diff
@@ -10,6 +16,12 @@ export function getSubmissionSamples(submission?: SubmissionMetadata) {
   if (!sampleDataField) {
     return [];
   }
+  if (
+    !(sampleDataField in submission.metadata_submission.sampleData) &&
+    options.createSampleDataFieldIfMissing
+  ) {
+    submission.metadata_submission.sampleData[sampleDataField] = [];
```

Comment:
The only thing I don't understand about this change is why the default value is an array (and not something that depends upon the _type_ associated with whatever `submission.metadata_submission.sampleData[sampleDataField]` is. I think this is just a matter of me not knowing about the latter variable. I'll chalk this up to that—and I'll make myself a note to learn about the latter (for one thing, so I can more effectively review code).

Reply:
It's a fair question. It might be helpful to think about the structure of `sampleData` in terms of the desktop submission portal's interface. 

![image](https://github.com/microbiomedata/nmdc-field-notes/assets/102547565/90578151-816a-4cb6-9b8b-2e597396a90c)

The keys of `sampleData` correspond to the tabs (e.g. the "Soil" tab corresponds to the `soil_data` key in `sampleData`). Within the "Soil" tab/`soil_data` key the array elements correspond to rows in the spreadsheet -- what we refer to as individual "samples". So the underlying data corresponding to that screenshot looks something like:

```js
{
  metadata_submission: {
    sampleData: {
      soil_data: [
        { samp_name: "samp 1", ... },  // <- this is a sample
        { samp_name: "samp 2", ... },  // <- this is another sample
        { samp_name: "samp 3", ... }, 
        { samp_name: "samp 4", ... }, 
      ],
      emsl_data: [ 
        ... // this is stuff we'll never show in the app
      ],
      jgi_mg_data: [ 
        ... // this is stuff we'll never show in the app
      ],
      jgi_mt_data: [ 
        ... // this is stuff we'll never show in the app
      ],
    }
  }
}
```

Reply:
Ah, OK, thanks! It makes sense to me now. Thanks for providing the visual aids.